api/lib/domain/usecases/generate-username-with-temporary-password.js

```diff
@@ -18,12 +18,18 @@ module.exports = async function generateUsernameWithTemporaryPassword({
   _checkIfStudentAccountAlreadyHasUsername(studentAccount);
 
   const username = await userReconciliationService.createUsernameByUser({ user: schoolingRegistration , userRepository });
-  const generatedPassword = passwordGenerator.generate();
-  const hashedPassword = await encryptionService.hashPassword(generatedPassword);
 
-  await userRepository.updateUsernameAndPassword(studentAccount.id, username, hashedPassword);
+  if (studentAccount.password) {
+    const updatedUser = await userRepository.addUsername(studentAccount.id, username);
+    return { username: updatedUser.username, password: undefined  };
```

Comment:
```suggestion
    return { username: updatedUser.username  };
```
ou 
```suggestion
    return { username: updatedUser.username, generatedPassword: undefined  };
```
Renvoyer un attribut `password` ici ne sert à rien car, si on regarde l'usage qui est fait de ce retour dans le serailizer `schooling-registration-dependent-user-serializer.js`, seuls les attributs `username` et `generatedPassword` sont utlisés. Personnellement, je préfère la première proposition.

Reply:
J'avait gardé un password `undefined` pour bien différencier et expliciter l'intention, mais si c'est clair dans le contexte, la première proposition de @MathieuGilet  est OK pour moi !

Reply:
Ok pour la première proposition !